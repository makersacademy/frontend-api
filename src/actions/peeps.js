import store from '../store/configureStore'

export const addPeep = (peep) => ({
  type: 'ADD_PEEP',
  peep
})

export const startAddPeep = (peepData = {}) => {
  return (dispatch) => {
    const session = store.getState().sessions
    const data = new FormData()
    const requestHeader = {
      method: "POST",
      headers: new Headers({
        'Authorization': `Token token=${session.session_key}`,
        'Content-Type': 'application/json'
      }),
      body: data.append("json", JSON.stringify({"peep": {"user_id": session.user_id, ...peepData}))
    }
    return fetch("https://chitter-backend-api.herokuapp.com/peeps", requestHeader).then(response => {
      return response.json()
    }).then(json => {
      dispatch(addPeep(json))
    }).catch(e => console.log(e))
  }
}

export const removePeep = ({id} = {}) => ({
  type: 'REMOVE_PEEP',
  id
})

export const setPeeps = (peeps) => ({
  type: 'SET_PEEPS',
  peeps
})

export const startSetPeeps = () => {
  return (dispatch) => {
    return fetch("https://chitter-backend-api.herokuapp.com/peeps").then(response => {
      return response.json()
    }).then(json => {
      dispatch(setPeeps(json))
    }).catch((e) => console.log(e))
  }
}
