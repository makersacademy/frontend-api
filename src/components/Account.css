.ui.card {
  margin: auto !important;
  width: 100% !important;
  height: 100% !important;
}

div.signout.extra.content a {
  color: #4183c4 !important;
}