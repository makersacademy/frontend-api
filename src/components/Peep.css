#root, html, body {
  height: 100vh !important;
}

.app {
  height: 100%;
}

.top-menu {
  height: 10vh !important;
  margin: 0 !important;
  padding: 0 !important;
}

.content-field {
  margin-top: 5% !important;
  height: 85% !important;
}

.new-peep .field {
  height: 100%;
}

.new-peep {
  height: 100% !important;
}

.new-peep i,
.new-peep button {
  position: fixed !important;
  bottom: 10% !important;
  right: 10% !important;
  padding: 0 !important;
  margin: 0 !important;
  border: none !important;
  opacity: 0.8 !important;
  cursor: pointer !important;
  z-index: 1 !important;
}

.new-peep i:hover {
  opacity: 1 !important;
}

textarea {
  max-height: 100% !important;
  font-size: 2rem !important;
  height: 100% !important;
  resize: none !important;
  border: none !important;
}