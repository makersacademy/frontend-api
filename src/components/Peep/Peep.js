import React from 'react';
import Classes from './Peep.module.css';
import axios from '../../axios-chitter';

class Peep extends React.Component {
    state = {
        isLuvd: false,
    }

    toggleLuv = (e) => {
        e.preventDefault();
        this.setState(prevState => ({ isLuvd: !prevState.isLuvd }));
    }


    render() {
        const options = { 
            weekday: "short", year: "numeric", month: "short", 
            day: "numeric", hour: "2-digit", minute: "2-digit"
        };

        const peepDate = new Date(this.props.time).toLocaleString('en-us', options)
        const luvs = this.props.luvs === 1 ? 'Luv' : 'Luvs';
        const blueLuvHeart = '../../../images/blu-luv-heart.png';
        const pinkLuvHeart = '../../../images/pink-luv-heart.png';
        let image = <img 
            src={blueLuvHeart} 
            height='13px' 
            alt='a luv icon'
            data-test='luv-icon' />
        
        if (this.state.isLuvd) image = <img 
            src={pinkLuvHeart} 
            height='13px' 
            alt='a luv icon'
            data-test='luv-icon' />

        return (
            <div className={Classes.Peep} data-test='component-peep'>
                <div>
                    <div className={Classes.Header}>
                        <img 
                            src={this.props.image} 
                            alt='a chitter user' 
                            data-test='user-image'
                            height='20px' />
                        <div className={Classes.Handle} data-test='user-handle'>@{this.props.handle}</div>
                    </div>
                    <div className={Classes.PeepBody} data-test='peep-body'>{this.props.body}</div>
                    <div className={Classes.Footer}>
                        <div className={Classes.FooterLeft}><div data-test='peeped-on'>Peeped on {peepDate}</div></div>
                        <div className={Classes.FooterRight}>
                            <div onClick={this.toggleLuv} data-test='number-of-luvs'>{this.props.luvs} {luvs}</div>
                            <a href='/#' onClick={(e) => this.toggleLuv(e)}>{image}</a>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default Peep;
