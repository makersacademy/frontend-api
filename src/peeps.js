$(document).ready(function(){
  $.get('https://chitter-backend-api.herokuapp.com/peeps', function(data) {
    data.forEach(function(element) {
      $(".peeps").append("<li>" + element.body + "</li>");
    });

    
  });

  // $(document).ready(function(){
  //   $("#btn1").click(function(){
  //     $("p").append(" <b>Appended text</b>.");
  //   });
  //
  //   $("#btn2").click(function(){
  //     $("ol").append("<li>Appended item</li>");
  //   });
  // });
  //
});
