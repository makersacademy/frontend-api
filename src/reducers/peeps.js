const peepsReducerDefaultState = [];

export default (state=peepsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_PEEP':
      return [action.peep, ...state]
    case 'REMOVE_PEEP':
      return state.filter(({id}) => id !== action.id)
    case 'SET_PEEPS':
      return action.peeps
    case 'LIKE_PEEP':
      state[state.findIndex((peep) => peep.id === action.id)].likes.push(action.user)
      return state
    case 'UNLIKE_PEEP':
      state[state.findIndex((peep) => peep.id === action.id)].likes = []
      return state
    default:
      return state
  }
}
