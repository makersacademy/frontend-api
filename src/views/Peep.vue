<template>
  <div class="title">
    <h1>Chitter</h1>
    <h2>{{$route.params.id}}</h2>
    <div class="peep">
        here
        {{$route.params.id}}
        <!-- <h2>{{peep.body}}</a></h2>
        <h4>{{peep.user.handle}}</h4> -->
      </div>
    </div>
  </div>
</template>

<script>
import axios from 'axios'
import router from '../router'

// const BASE_URL = 'https://chitter-backend-api.herokuapp.com/'
const BASE_URL = 'http://localhost:3000/'

export default {
  // name: 'Peep',
  // data: function(){
  //   return {
  //     peep: []
  //   }
  // },
  // created: function(){
  //   var self = this
  //   console.log('triggered1')
  //   self.getpeep()
  // },
  // methods: {
  //         getpeep() {
  //             var self = this
  //             console.log('triggered2')
  //             axios.get(BASE_URL + 'peeps/1')
  //             .then(function (response) {
  //               console.log(response)
  //               self.peeps = response.data;
  //             })
  //             .catch(function (error) {
  //               console.log(error);
  //             })
  //           }
  // }

}
</script>

<style>
img {
  max-width: 50vw;
}
</style>