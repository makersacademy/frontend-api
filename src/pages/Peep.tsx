import { useParams } from "react-router-dom";
import { peepType } from "../../types/apiData";
import { Loader } from "../components/Loader";
import { useGlobalContext } from "../Context/globalContext";
import { useFetch } from "../hooks/useFetch";

export const Peep = () => {
  const { id } = useParams();
  const { client } = useGlobalContext();
  const [data, isLoading, isError, error] = useFetch<peepType>({
    querykey: { peepId: id },
    queryFn: client.findPeepById,
  });

  return (
    <>
      {isLoading && <Loader />}
      {isError && <h1>{error}</h1>}
      {!isLoading && data && <PeepContent {...data} />}
    </>
  );
};

const PeepContent = ({ id, body, user }: peepType) => {
  return (
    <>
      <pre>{id}</pre>
      <h1>{user.handle}</h1>
      <p>{body}</p>
    </>
  );
};
