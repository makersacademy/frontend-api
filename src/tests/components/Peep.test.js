import React from 'react';
import { shallow } from 'enzyme';
import { Peep } from '../../components/Peep'
import peeps from '../fixtures/peeps'

let wrapper

beforeEach(() => {
  wrapper = shallow(<Peep peep={peeps[0]}/>)
})

test('should render Feed correctly', () => {
  expect(wrapper).toMatchSnapshot();
})
