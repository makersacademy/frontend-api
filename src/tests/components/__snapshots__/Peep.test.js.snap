// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Feed correctly 1`] = `
<div
  className="border"
>
  <p>
    #
    toddpla
     @ 
    2018-11-17T19:40:58.497Z
  </p>
  <p>
    test peep one
  </p>
  <Connect(Like)
    id={1}
    likes={Array []}
  />
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
